bundles/pubsub/pubsub_admin_zmq/src/pubsub_zmq_topic_sender.c

```diff
@@ -635,13 +631,24 @@ static int psa_zmq_topicPublicationSend(void* handle, unsigned int msgTypeId, co
 
                     //send MetaData
                     if (rc > 0 && metadataLength > 0) {
-                        zmq_msg_init_data(&msg3, metadataData, metadataLength, psa_zmq_freeMsg, freeMsgEntry);
-                        rc = zmq_msg_send(&msg3, socket, 0);
+                        zmq_msg_init_data(&msg3, metadataData, metadataLength, psa_zmq_freeMsg, NULL);
```

Comment:
Because this is not correct.
According zmq_msg_init_data, the zmq_free_fn *ffn is called when the data is not needed anymore. The freeMsgEntry is hint to remove serialized output of the payload.
When the metadata is not needed anymore the metadata should be removed, not the serialized data.

Reply:
I agree, let's see what it does in usage.